acceptance-tests/src/test/java/tech/pegasys/eth2signer/dsl/lotus/LotusNode.java

```diff
@@ -34,63 +34,48 @@
  */
 public class LotusNode {
   private static final String FC_URL_FORMAT = "http://%s:%d/rpc/v0";
+
   private final String fcUrl;
   private final JsonRpcClient jsonRpcClient;
+  public static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().registerModule(new FilecoinJsonRpcModule());
 
   public LotusNode(final String host, final int port) {
     fcUrl = String.format(FC_URL_FORMAT, host, port);
-    jsonRpcClient = initClient();
+    jsonRpcClient =
+        new JsonRpcClient(
+            request -> FilecoinJsonRequests.executeRawJsonRpcRequest(fcUrl, request),
+            OBJECT_MAPPER);
   }
 
   public LotusNode(final int port) {
     this("127.0.0.1", port);
   }
 
-  public void loadDefaultAddresses() {
-    AddressesUtil.getDefaultFilecoinAddressMap()
-        .forEach(
-            (address, key) -> {
-              if (!hasAddress(address)) {
-                importPrivateKey(key);
-              }
-            });
-  }
+  public Map<String, FilecoinKey> loadAddresses(final int blsKeys, final int secpKeys) {
+    final Set<String> addresses = new HashSet<>();
+    final Map<String, FilecoinKey> addressKeys = new HashMap<>();
 
-  public JsonRpcClient getJsonRpcClient() {
-    return jsonRpcClient;
-  }
+    for (int i = 0; i < blsKeys; i++) {
+      final String address = FilecoinJsonRequests.walletNew(jsonRpcClient, BLS_SIGTYPE);
+      addresses.add(address);
+    }
 
-  private void importPrivateKey(final FilecoinKey key) {
-    jsonRpcClient
-        .createRequest()
-        .method("Filecoin.WalletImport")
-        .params(key)
-        .id(202)
-        .returnAs(String.class)
-        .execute();
-  }
+    for (int i = 0; i < secpKeys; i++) {
+      final String address = FilecoinJsonRequests.walletNew(jsonRpcClient, SECP_SIGTYPE);
+      addresses.add(address);
+    }
 
-  public Boolean hasAddress(final String address) {
-    return jsonRpcClient
-        .createRequest()
-        .method("Filecoin.WalletHas")
-        .params(address)
-        .id(101)
-        .returnAs(Boolean.class)
-        .execute();
-  }
+    addresses.forEach(
+        address -> {
+          final FilecoinKey filecoinKey = FilecoinJsonRequests.walletExport(jsonRpcClient, address);
+          addressKeys.put(address, filecoinKey);
+        });
 
-  public String executeRawJsonRpcRequest(final String request) throws IOException {
-    final HttpPost post = new HttpPost(fcUrl);
-    post.setEntity(new StringEntity(request, Charsets.UTF_8));
-    post.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.JSON_UTF_8.toString());
-    try (final CloseableHttpClient httpClient = HttpClients.createDefault();
-        final CloseableHttpResponse httpResponse = httpClient.execute(post)) {
-      return EntityUtils.toString(httpResponse.getEntity(), Charsets.UTF_8);
-    }
+    return Map.copyOf(addressKeys);
```

Comment:
why returning a copy of the map? Isn't it just on the stack?

Reply:
original idea was to return an unmodifiedable map copy. Now it is replaced by the stream call.